src/QsCompiler/Transformations/SearchAndReplace.cs

```diff
@@ -222,6 +222,26 @@ public IdentifierReferences(QsQualifiedName idName, QsLocation? defaultOffset, I
             return finder.SharedState.Locations;
         }
 
+        /// <summary>
+        /// Finds references to an identifier in an expression.
+        /// </summary>
+        /// <param name="name">The identifier name.</param>
+        /// <param name="expr">The expression.</param>
+        /// <param name="fileName">The source file name.</param>
+        /// <param name="specPosition">The position of the specialization that contains the expression.</param>
+        /// <param name="statementLocation">The location of the statement that contains the expression.</param>
+        /// <returns>The locations of the references.</returns>
+        public static ImmutableHashSet<Location> FindInExpression(
```

Comment:
Because the find methods have so many parameters, I think overloading them is confusing - it's difficult to tell what each one is doing just by looking at the parameter list. I would rather rename the other finds to FindInScope and FindInNamespace.

Reply:
I am fine either way - I am also ok deprecating the finds and naming them more specifically. I have a slight preference towards keeping some consistency.

Reply:
I replaced one of the `Find`s with `FindInScope`. For the namespace `Find`, since it's only 3 lines and didn't quite fit the pattern of the other ones (because of the out parameter), I replaced its only use with code to directly use the transformation instance instead.